File: server/medicore.layer/MediCore.Service/Model/HospitalStaffOperatingHoursModel.cs
using dna.core.service.Models.Abstract;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace MediCore.Service.Model
{
    public class HospitalStaffOperatingHoursModel : WriteHistoryBaseModel, IModelBase
    {
        public int Id { get; set; }

        [Required(ErrorMessage = "Required")]
        public int Day { get; set; }

        [Required(ErrorMessage = "Required")]
        public int HospitalMedicalStaffId { get; set; }

        [Required(ErrorMessage = "Required")]
        [StringLength(10, ErrorMessage = "StringLength")]
        public string StartTime { get; set; }

        [Required(ErrorMessage = "Required")]
        [StringLength(10, ErrorMessage = "StringLength")]
        public string EndTime { get; set; }

        [Required(ErrorMessage = "Required")]
        public bool IsClossed { get; set; }

        public virtual HospitalMedicalStaffModel HospitalStaff { get; set; }

    }
}

File: dashboard/src/app/model/hospitalStaffOperatingHoursModel.ts
import { IEntityBase } from './IEntityBase';
import { WriteHistoryBaseModel } from './writeHistoryBaseModel';
import { HospitalModel } from './hospitalModel';
import { HospitalMedicalStaffModel } from './hospitalMedicalStaffModel';

export class HospitalStaffOperatingHoursModel extends WriteHistoryBaseModel implements IEntityBase{
    id:number;
    day: number;
    hospitalMedicalStaffId: number;
    startTime: string;
    endTime: string;
    isClossed: boolean;
    hospitalStaff: HospitalMedicalStaffModel;

    constructor()
    constructor(day:number, isClossed:boolean)
    constructor(day:number, isClossed:boolean, startTime: string, endTime: string)
    constructor(day?:number, isClossed?:boolean, startTime?:string, endTime?:string){       
        super();
        this.day = day;
        this.startTime = startTime;
        this.endTime = endTime;        
        this.isClossed = isClossed;
        this.hospitalStaff = null;
       
    }
}